Userland/Libraries/LibWeb/Fetch/Fetching/Fetching.cpp

```diff
@@ -1658,6 +1659,14 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<PendingResponse>> nonstandard_resource_load
                 auto header = TRY_OR_IGNORE(Infrastructure::Header::from_string_pair(name, value));
                 TRY_OR_IGNORE(response->header_list()->append(header));
             }
+
+            // NOTE: For file:// URLs, we have to guess the MIME type, since there's no HTTP header to tell us what this is.
+            if (request->url().scheme() == "file"sv) {
+                auto mime_type = Core::guess_mime_type_based_on_filename(request->url().path());
+                auto header = TRY_OR_IGNORE(Infrastructure::Header::from_string_pair("Content-Type"sv, mime_type));
+                TRY_OR_IGNORE(response->header_list()->append(move(header)));
+            }
```

Comment:
Perhaps we should add this header in ResourceLoader instead as presumably other code loading `file://` URLs through it does similar things?